sdk/azidentity/default_azure_credential.go

```diff
@@ -51,33 +53,44 @@ func NewDefaultAzureCredential(options *DefaultAzureCredentialOptions) (*Default
 	if err == nil {
 		creds = append(creds, envCred)
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("EnvironmentCredential: %s", err.Error()))
 	}
 
 	msiCred, err := NewManagedIdentityCredential(&ManagedIdentityCredentialOptions{ClientOptions: options.ClientOptions})
 	if err == nil {
 		creds = append(creds, msiCred)
 		msiCred.client.imdsTimeout = time.Second
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("ManagedIdentityCredential: %s", err.Error()))
 	}
 
 	cliCred, err := NewAzureCLICredential(&AzureCLICredentialOptions{TenantID: options.TenantID})
 	if err == nil {
 		creds = append(creds, cliCred)
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("AzureCLICredential: %s", err.Error()))
+	}
+
+	errMsg := "\n"
+	for _, msg := range errorMessages {
+		errMsg += fmt.Sprintf("\t%s\n", msg)
 	}
 
 	if len(creds) == 0 {
 		err := errors.New(errMsg)
 		logCredentialError("Default Azure Credential", err)
 		return nil, err
 	}
+
+	if len(errMsg) != 0 {
```

Comment:
`len("\n") == 1`

Reply:
I had it empty before, then I forgot to check 😆 thanks